Update Backend Challenge (#36)
gravitational/careers: challenges/systems/challenge.md

Before writing the actual code, create a brief design document in Google Docs or markdown in Github and share with the team.

> This document should cover at least: design approach, trade-offs, scope, proposed API, and security considerations. For example, you should have a section called "Security" with three subsections: "Transport Layer", "Authorization", and "Authentication".

Reviewer: ```suggestion
This document should cover at least: design approach, trade-offs, scope, proposed API, and security considerations. For security, please include considerations for transport layer, authentication, and authorization.
```
